config/pmd/ruleset.xml

```diff
@@ -1,325 +1,329 @@
+  <!-- <rule ref="category/java/design.xml/SwitchDensity" /> -->
+  <!-- <rule ref="category/java/design.xml/TooManyFields" /> -->
+  <!-- <rule ref="category/java/design.xml/TooManyMethods" /> -->
+  <rule ref="category/java/design.xml/UselessOverridingMethod" />
+  <!-- <rule ref="category/java/design.xml/UseObjectForClearerAPI" /> -->
+  <!-- <rule ref="category/java/design.xml/UseUtilityClass"/> -->
 
 
-    <!-- <rule ref="category/java/documentation.xml/CommentContent" /> -->
-    <!-- <rule ref="category/java/documentation.xml/CommentRequired" /> -->
-    <!-- <rule ref="category/java/documentation.xml/CommentSize" /> -->
-    <rule ref="category/java/documentation.xml/UncommentedEmptyConstructor"/>
-    <rule ref="category/java/documentation.xml/UncommentedEmptyMethodBody"/>
+  <!-- <rule ref="category/java/documentation.xml/CommentContent" /> -->
+  <!-- <rule ref="category/java/documentation.xml/CommentRequired" /> -->
+  <!-- <rule ref="category/java/documentation.xml/CommentSize" /> -->
+  <rule ref="category/java/documentation.xml/UncommentedEmptyConstructor" />
+  <rule ref="category/java/documentation.xml/UncommentedEmptyMethodBody" />
 
 
+  <rule ref="category/java/errorprone.xml/AssignmentInOperand">
+    <properties>
+      <property name="allowWhile" value="true" />
+    </properties>
+  </rule>
+  <rule ref="category/java/errorprone.xml/AssignmentToNonFinalStatic" />
+  <rule ref="category/java/errorprone.xml/AvoidAccessibilityAlteration" />
+  <!-- <rule ref="category/java/errorprone.xml/AvoidAssertAsIdentifier" /> -->
+  <rule ref="category/java/errorprone.xml/AvoidBranchingStatementAsLastInLoop" />
+  <!-- <rule ref="category/java/errorprone.xml/AvoidCallingFinalize" /> -->
+  <!-- <rule ref="category/java/errorprone.xml/AvoidCatchingNPE" /> -->
+  <rule ref="category/java/errorprone.xml/AvoidCatchingThrowable" />
+  <rule ref="category/java/errorprone.xml/AvoidDecimalLiteralsInBigDecimalConstructor" />
+  <!-- <rule ref="category/java/errorprone.xml/AvoidDuplicateLiterals" /> -->
+  <!-- <rule ref="category/java/errorprone.xml/AvoidEnumAsIdentifier" /> -->
+  <!-- <rule ref="category/java/errorprone.xml/AvoidFieldNameMatchingMethodName" /> -->
+  <!-- <rule ref="category/java/errorprone.xml/AvoidFieldNameMatchingTypeName" /> -->
+  <rule ref="category/java/errorprone.xml/AvoidInstanceofChecksInCatchClause" />
+  <!-- <rule ref="category/java/errorprone.xml/AvoidLiteralsInIfCondition" /> -->
+  <!-- <rule ref="category/java/errorprone.xml/AvoidLosingExceptionInformation" /> -->
+  <rule ref="category/java/errorprone.xml/AvoidMultipleUnaryOperators" />
+  <rule ref="category/java/errorprone.xml/AvoidUsingOctalValues" />
+  <rule ref="category/java/errorprone.xml/BadComparison" />
+  <rule ref="category/java/errorprone.xml/BeanMembersShouldSerialize">
+    <properties>
+      <property name="ignoredAnnotations"
+        value="org.springframework.beans.factory.annotation.Autowired|lombok.Data|lombok.Getter|lombok.Value" />
```

Comment:
Formater tutaj poszalał, ale generalnie to się pozbyło problemu z `@Autowired` - tylko trzeba było podbić wersję, bo w starszej chyba nie ma